src/quartz/scene/doodad/Doodad.cpp

```diff
@@ -27,21 +27,26 @@ quartz::scene::Doodad::Doodad(
     const quartz::rendering::Device& renderingDevice,
     quartz::managers::PhysicsManager& physicsManager,
     std::optional<quartz::physics::Field>& o_field,
-    const std::string& objectFilepath,
+    const std::optional<std::string>& o_objectFilepath,
     const math::Transform& transform,
-    const std::optional<quartz::physics::RigidBody::Parameters>& o_rigidBodyParameters
+    const std::optional<quartz::physics::RigidBody::Parameters>& o_rigidBodyParameters,
+    const quartz::scene::Doodad::FixedUpdateCallback& fixedUpdateCallback,
+    const quartz::scene::Doodad::UpdateCallback& updateCallback
 ) :
-    m_model(
-        renderingDevice,
-        objectFilepath
+    mo_model(
+        o_objectFilepath ?
+            std::optional<quartz::rendering::Model>(quartz::rendering::Model(renderingDevice, *o_objectFilepath)) :
+            std::nullopt
     ),
     m_transform(quartz::scene::Doodad::fixTransform(transform)),
     m_transformationMatrix(),
     mo_rigidBody(
         (o_field && o_rigidBodyParameters) ?
             std::optional<quartz::physics::RigidBody>(o_field->createRigidBody(physicsManager, m_transform, *o_rigidBodyParameters)) :
             std::nullopt
-    )
+    ),
+    m_fixedUpdateCallback(fixedUpdateCallback),
+    m_updateCallback(updateCallback)
```

Comment:
Perhaps we can check if the provided callbacks are bad, and if they are bad then we can create our own no-op callback. This will guarantee that the callbacks will be valid so we don't need to encapsulate them in a conditional within the update functions